lib/archethic_web/controllers/faucet_controller.ex

```diff
@@ -40,22 +40,22 @@ defmodule ArchEthicWeb.FaucetController do
   def create_transfer(conn, %{"address" => address}) do
     with {:ok, recipient_address} <- Base.decode16(address, case: :mixed),
          true <- Crypto.valid_hash?(recipient_address),
-         :ok <- transfer(recipient_address) do
+         {tx_address, :ok} <- transfer(recipient_address) do
       conn
       |> put_resp_header("cache-control", "no-cache, no-store, must-revalidate")
       |> put_resp_header("pragma", "no-cache")
       |> put_resp_header("expires", "0")
       |> put_flash(:info, "Transferred successfully (click to view)")
-      |> render("index.html", address: "", link_address: address)
+      |> render("index.html", address: "", link_address: tx_address)
     else
-      {:error, _} ->
+      e when e in [:error, false] ->
```

Comment:
I think you should keep the previous checking, with `{:error, _}` as before and use `_` for the malformed address, as it will be more explicit and easy to understand

Reply:
I discovered that in a "network_error" event, I was getting `Malformed address` error using the earlier approach.

Reply:
During that error the e value was like this 
> {"0098fe10e8633bce19c59a40a089731c1f72b097c5a8f7dc71a37eb26913aa4f80",{:error,....}}